chain/get_ancestors.go

```diff
@@ -124,3 +127,43 @@ func CollectAtMostNTipSets(ctx context.Context, iterator *TipsetIterator, n uint
 	}
 	return ret, nil
 }
+
+// FindCommonAncestor returns the common ancestor of the two tipsets pointed to
+// by the input iterators.  If they share no common ancestor errIterComplete
+// will be returned.
+func FindCommonAncestor(oldIter, newIter *TipsetIterator) (types.TipSet, error) {
+	for !newIter.Complete() && !oldIter.Complete() {
+		old := oldIter.Value()
+		new := newIter.Value()
+
+		oldHeight, err := old.Height()
+		if err != nil {
+			return nil, err
+		}
+		newHeight, err := new.Height()
+		if err != nil {
+			return nil, err
+		}
+
+		// Found common ancestor.
+		if old.Equals(new) {
+			return old, nil
+		}
+
+		// Update the pointers.  Pointers move back one tipset if they
+		// point to a tipset at the same height or higher than the
+		// other pointer's tipset.
+		if newHeight >= oldHeight {
+			if err := newIter.Next(); err != nil {
+				return nil, err
+			}
+		}
+
+		if newHeight < oldHeight {
```

Comment:
Blocking: why not `<=`? To make the symmetry obvious, I recommend mirroring L156: `if oldHeight >= newHeight`.

If I've misunderstood something that requires this asymmetry, it definitely needs a comment.

Reply:
`oldHeight >= newHeight` is better (though not strictly necessary).  Thanks for catching